networkx/generators/classic.py

```diff
@@ -526,25 +535,30 @@ def dorogovtsev_goltsev_mendes_graph(n, create_using=None):
     References
     ----------
     .. [1] S. N. Dorogovtsev, A. V. Goltsev and J. F. F. Mendes,
-        "Pseudofractal scale-free web", Physical Review E 65, 066122, 2002.
+        "Pseudofractal scale-free web," Physical Review E 65, 066122, 2002.
```

Comment:
Shouldn't this comma go outside the quote in a citation? Why change it?

Reply:
I made the change because I feel like I've only ever seen commas and periods _inside_ the quote in citations.

Beyond my personal experience, I had a (quick) look and found several guidelines for reference lists that either use punctuation inside quotation marks, or no quotation marks at all: [Chicago](https://pitt.libguides.com/citationhelp/chicago), [APA](https://pitt.libguides.com/citationhelp/apa7), [IEEE](https://pitt.libguides.com/citationhelp/ieee).

I don't know if there is an exact style you want to adhere to with NetworkX citations. If there is, I'm quite sure it's not being followed properly right now, and I'd be open to going over the codebase and making references conform to a chosen standard in a separate PR.

Reply:
The title does not include a comma.  And the quotes are around the title.
The commas separate the different parts of the citation. They should not include the quote mark as part of the next section of the citation. I can't believe there is a style which puts the commas that indicate sections of the citation inside the quotes that enclose the title. (but I could be wrong. :)

Reply:
I'm aware. Did you look at the links I provided? They have some similar examples where I believe the comma serves the same delimiting purpose, and is placed inside the quotation marks.

Reply:
I do see that and while my confidence in citation styles has always been low, it is much lower now. They don't seem to respect the significance of punctuation as separators.   (pouts sulkily to the side) 

There does seem to be [some conversation if not debate](https://style.mla.org/punctuation-and-quotation-marks/) about this issue, including differences in British and US conventions.

I'd really prefer that it be left outside the quotes in our code...

Reply:
The funny thing is, I actually agree with you. It makes no sense semantically to include commas and periods in the thing you're quoting. To emphasize this, the "punctuation outside quotes" style is often referred to as "logical punctuation," (ugh...) and like you mention is the one used in British English.

I feel like I have to cut the citation styles some slack though; punctuation inside quotes isn't something they came up with, but rather an extrapolation of the existing rule in formal writing. It also only seems to apply to commas and periods, for some reason; other punctuation gets treated logically.

This prompted me to go down a small rabbit hole to try to find an answer to the potentially more interesting question of _why_ this ever became a rule in the first place.
> If it seems hard or even impossible to defend the American way on the merits, that’s probably because it emerged from aesthetic, not logical, considerations. According to Rosemary Feal, executive director of the MLA, it was instituted in the early days of the Republic in order “to improve the appearance of the text. A comma or period that follows a closing quotation mark appears to hang off by itself and creates a gap in the line (since the space over the mark combines with the following word space).” I don’t doubt Feal, but the appearance argument doesn’t carry much heft today; more to the point is that we are simply accustomed to the style.

The more I think about it, the more I believe logic should trump tradition. I'll change the punctuation like you're suggesting. I do want to emphasize the importance of consistency, no matter what choice we end up going with. If you provide me with a standardized format, I'll gladly go over the existing references in the code to ensure we conform to it.

PS: You'll be delighted to reread [this](https://github.com/networkx/networkx/issues/7522#issuecomment-2191849959) comment of yours, made earlier today! /s

Reply:
Hee hee.... yes, in [my comment earlier today](https://github.com/networkx/networkx/issues/7522#issuecomment-2191849959), I did actually cut/paste the citation with a period inside the quote...   But I would argue that the period is not separating the title from the date in that case. Parentheses are used for that. The period seems to be the end of the title written as a sentence. I did a double-take when copy/pasting to see if it really was a sentence. and then went on to other things. :)

Reply:
I don't doubt that; I'm only bringing it up jokingly, hence the sarcasm tag (/s).